bears/general/KeywordBear.py

```diff
@@ -2,18 +2,71 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
 
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.results.SourceRange import SourceRange
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return []
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return []
+    comments = []
+    for result in annotation_bear_results:
+        if not isinstance(result.contents, dict):
+            continue
+        comments += list(result.contents.get('comments', []))
+    return set(comments)
+
+
+def get_diff_comment_todo(dependency_results, file, filename,
+                          line, line_number, pos):
+    comments = _get_comments(dependency_results)
+    diffs = None
+    for source_range_comment in comments:
+        todo_source_range = SourceRange.from_values(
+            filename, line_number + 1, pos + 1)
```

Comment:
can we do this outside the loop? Also I wouldn't name it todo_source_range because this bear isn't necessarily only for TODOs but for anything basically